packages/go_router_builder/example/lib/stateful_shell_route_initial_location_example.g.dart

```diff
@@ -69,7 +69,7 @@ extension $NotificationsRouteDataExtension on NotificationsRouteData {
   static NotificationsRouteData _fromState(GoRouterState state) =>
       NotificationsRouteData(
         section: _$NotificationsPageSectionEnumMap
-            ._$fromName(state.pathParameters['section']!),
+            ._$fromName(state.pathParameters['section'] ?? ''),
```

Comment:
path parameters probably shouldn't be nullable?

Reply:
it's definitely not null, but firebase got a `!` error so instead of using it, I replaced it with `??` to solve problems

Reply:
what I meant is whether the code gen should only generate `??` when this is a query parameter.

However, I still not sure what would be store in the ProductRouter.id if query parmeter is null. I think we should force the ProductRouter.id to have a default value if it is not null and is used as query parmeter.

Reply:
@ho-doan WDYT?